core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -613,33 +486,13 @@ public void read(DataInput in) throws IOException {
             fsin, this.region.getOffset(), this.region.getCompressedSize());
         InputStream inputStreamToBeCompressed = boundedRangeFileInputStream;
 
-        if (cryptoParams != null && cryptoModule != null) {
-          DataInputStream tempDataInputStream = new DataInputStream(boundedRangeFileInputStream);
-          // Read the init vector from the front of the stream before initializing the cipher stream
-
-          int ivLength = tempDataInputStream.readInt();
-          byte[] initVector = new byte[ivLength];
-          tempDataInputStream.readFully(initVector);
-
-          cryptoParams.setInitializationVector(initVector);
-          cryptoParams.setEncryptedInputStream(boundedRangeFileInputStream);
-
-          // These two flags mirror those in WBlockState, and are very necessary to set in order
-          // that the underlying stream be written and handled
-          // correctly.
-          cryptoParams.setCloseUnderylingStreamAfterCryptoStreamClose(false);
-          cryptoParams.setRecordParametersToStream(false);
-
-          cryptoParams = cryptoModule.getDecryptingInputStream(cryptoParams);
-          inputStreamToBeCompressed = cryptoParams.getPlaintextInputStream();
-        }
-
         try {
+          inputStreamToBeCompressed = encryptionStrategy.decryptStream(inputStreamToBeCompressed);
           this.in = compressAlgo.createDecompressionStream(inputStreamToBeCompressed, decompressor,
               getFSInputBufferSize(conf));
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
Also a bit overly broad catch type (at least it's not Throwable! :smiley_cat: ). Perhaps encryptStream and decryptStream should be more explicit about the checked exceptions they can throw?